app/models/miq_report/search.rb

```diff
@@ -99,6 +100,13 @@ def paged_view_search(options = {})
     else
       targets = db_class
     end
+
+    unless selected_ids.nil?
+      targets = targets.select do |one_target|
```

Comment:
@karelhala `targets` here can be only array of ids or model class. Is that right ? so we can use sql query.

what do you think about this ? 
`
targets = targets.first.kind_of?(Integer) ? targets & selected_ids : targets.where(:id => selected_ids)
`